### PR TITLE
mzcompose: Support capture_and_print

### PR DESCRIPTION
**Pull request by def-**
See discussion in https://materializeinc.slack.com/archives/C01LKF361MZ/p1706740088673749

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: misc/python/materialize/mzcompose/composition.py [hunk 1]
```diff
@@ -664,8 +697,8 @@ def run_testdrive_files(
             "testdrive",
             *args,
             rm=rm,
-            # needed for sufficient error information in the junit.xml
-            capture_stderr=True,
+            # needed for sufficient error information in the junit.xml while still printing to stdout during execution
+            capture_and_print=True,
```
Comment (nrainer-materialize) [inline misc/python/materialize/mzcompose/composition.py:701]: You might also want to adjust `def testdrive(...)` at around line 1081?

File: test/testdrive/bytes.td [hunk 2]
```diff
@@ -60,7 +60,7 @@ named_col  false     bytea
 > SELECT * FROM data_offset
 data           offset
 ------------------------
-"\\xc2\\xa92"  1
+"\\xc2blablasomeerror\\xa92"  1
```
Comment (nrainer-materialize) [inline test/testdrive/bytes.td:63]: Slipped in? 😀

File: misc/python/materialize/mzcompose/composition.py [hunk 3]
```diff
@@ -303,28 +306,54 @@ def invoke(
...
-                    close_fds=False,
-                    check=check,
-                    stdout=stdout,
-                    stderr=stderr,
-                    input=stdin,
-                    text=True,
-                    bufsize=1,
-                )
+                if capture_and_print:
+                    with subprocess.Popen(
```
Comment (nrainer-materialize) [inline misc/python/materialize/mzcompose/composition.py:325]: Optional: Extract method?

**Review by nrainer-materialize — APPROVED** [review]
Thank you!!

File: test/testdrive/bytes.td (hunk 2 above)
Comment (def-) [reply test/testdrive/bytes.td:63]: Nope, that was for testing. Is this how it's supposed to look?
Screenshot 2024-02-01 at 15 09 52

Comment (nrainer-materialize) [reply test/testdrive/bytes.td]: Here are some examples how it looks like (not all of the are testdrive instances): https://buildkite.com/materialize/tests/builds/73749

However, I unfortunately don't have a comparison error but only syntax errors included.

Let me see if I can still find a build...

Comment (nrainer-materialize) [reply test/testdrive/bytes.td]: I haven't found one but I think it should be ok.

**Pull request merged**